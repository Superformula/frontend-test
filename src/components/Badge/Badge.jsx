import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { color, space, display } from 'styled-system';

import { Small, GridItem } from '..';

const Dot = styled.span`
  border-radius: 50%;
  height: ${props => props.theme.fontSizes[0]}
  width: ${props => props.theme.fontSizes[0]}
  ${color}
  ${space}
  ${display}
`;

const Badge = ({ statusColor, textTransform, children }) => (
  <GridItem
    alignItems="center"
    display="flex"
  >
    <Dot
      bg={statusColor}
      display="inline-block"
      mr={1}
    />
    {children
    && (
      <Small color="black54" textTransform={textTransform}>
        {children}
      </Small>
    )}
  </GridItem>
);

Badge.defaultProps = {
  textTransform: 'uppercase',
};

Badge.propTypes = {
  statusColor: PropTypes.string.isRequired,
  textTransform: PropTypes.string,
  ...color.PropTypes,
  ...space.PropTypes,
  ...display.PropTypes,
};

export default Badge;
