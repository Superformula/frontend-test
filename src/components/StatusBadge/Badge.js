import styled, { css } from 'styled-components';
import { COLORS } from 'consts/colors';

export const Badge = styled.span`
  display: inline-block;
  border-radius: 40px;
  padding: 0 6px;
  text-transform: uppercase;
  letter-spacing: 0.57px;
  line-height: 16px;
  font-size: 8px;
  font-weight: 500;
  ${({ isOpen }) => (isOpen ? OPEN : CLOSED)}
`;

const OPEN = css`
  background-color: ${COLORS.GREEN};
  color: ${COLORS.PRIMARY};
`;

const CLOSED = css`
  background-color: ${COLORS.RED};
  color: ${COLORS.WHITE};
`;

Badge.displayName = 'Badge';
