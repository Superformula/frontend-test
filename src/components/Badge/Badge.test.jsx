import React from 'react';
import { shallowWithTheme } from '../../helpers/tests/withTheme';
import Badge from './Badge';

const setUp = (color, textTransform) => (
  shallowWithTheme(
    <Badge statusColor={color} textTransform={textTransform}>test badge</Badge>,
  )).dive();

describe('Button Component', () => {
  let component;
  it('renders a red badge', () => {
    component = setUp('red', 'uppercase');
    expect(component).toMatchSnapshot();
  });
  it('renders a green badge', () => {
    component = setUp('green', 'lowercase');
    expect(component).toMatchSnapshot();
  });
});
