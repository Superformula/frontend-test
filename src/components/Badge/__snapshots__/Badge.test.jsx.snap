// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders a green badge 1`] = `
<styled.div
  alignItems="center"
  display="flex"
>
  <styled.span
    bg="green"
    display="inline-block"
    mr={1}
  />
  <styled.small
    color="black54"
    textTransform="lowercase"
  >
    test badge
  </styled.small>
</styled.div>
`;

exports[`Button Component renders a red badge 1`] = `
<styled.div
  alignItems="center"
  display="flex"
>
  <styled.span
    bg="red"
    display="inline-block"
    mr={1}
  />
  <styled.small
    color="black54"
    textTransform="uppercase"
  >
    test badge
  </styled.small>
</styled.div>
`;
